// mobile/src/auth/AuthContext.tsx
import React, { createContext, useState, useEffect, useContext } from 'react';
import { profileApi, getToken, clearToken } from './api';

interface User {
  id: string;
  nickname: string;
  wechatId: string;
  avatarUrl?: string;
  signature?: string;
  phone?: string;
  email?: string;
}

interface AuthContextType {
  user: User | null;
  loading: boolean;
  login: (token: string, userData: User) => void;
  logout: () => void;
}

const AuthContext = createContext<AuthContextType>({
  user: null,
  loading: true,
  login: () => {},
  logout: () => {},
});

export function AuthProvider({ children }: { children: React.ReactNode }) {
  const [user, setUser] = useState<User | null>(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    getToken().then(token => {
      if (token) {
        profileApi.get()
          .then(setUser)
          .catch(() => clearToken())
          .finally(() => setLoading(false));
      } else {
        setLoading(false);
      }
    });
  }, []);

  const login = (token: string, userData: User) => {
    import('./api').then(m => m.setToken(token));
    setUser(userData);
  };

  const logout = () => {
    clearToken();
    setUser(null);
  };

  return (
    <AuthContext.Provider value={{ user, loading, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
}

export function useAuth() {
  return useContext(AuthContext);
}
